lua/telescope/_extensions/file_browser/stats.lua

```diff
@@ -0,0 +1,95 @@
+local M = {}
+
+local SIZE_TYPES = { "", "K", "M", "G", "T", "P", "E", "Z" }
+local YEAR = os.date "%Y"
+
+local SIZE_HL = "TelescopePreviewSize"
+M.size = {
+  width = 5,
+  right_justify = true,
+  display = function(entry)
+    local size = entry.stat.size
+    for _, v in ipairs(SIZE_TYPES) do
+      local type_size = math.abs(size)
+      if type_size < 1024.0 then
+        if type_size > 9 then
+          return { string.format("%3d%s", size, v), SIZE_HL }
+        else
+          return { string.format("%3.1f%s", size, v), SIZE_HL }
+        end
+      end
+      size = size / 1024.0
+    end
+    return { string.format("%.1f%s", size, "Y"), SIZE_HL }
+  end,
+}
+
+local DATE_HL = "TelescopePreviewDate"
+M.date = {
+  width = 13,
+  right_justify = true,
+  display = function(entry)
+    local mtime = entry.stat.mtime.sec
+    if YEAR ~= os.date("%Y", mtime) then
+      return { os.date("%b %d  %Y", mtime), DATE_HL }
+    end
+    return { os.date("%b %d %H:%M", mtime), DATE_HL }
+  end,
+}
+
+local color_hash = {
+  ["d"] = "TelescopePreviewDirectory",
+  ["l"] = "TelescopePreviewLink",
+  ["s"] = "TelescopePreviewSocket",
+  ["r"] = "TelescopePreviewRead",
+  ["w"] = "TelescopePreviewWrite",
+  ["x"] = "TelescopePreviewExecute",
+  ["-"] = "TelescopePreviewHyphen",
+}
+
+local mode_perm_map = {
+  ["0"] = { "-", "-", "-" },
+  ["1"] = { "-", "-", "x" },
+  ["2"] = { "-", "w", "-" },
+  ["3"] = { "-", "w", "x" },
+  ["4"] = { "r", "-", "-" },
+  ["5"] = { "r", "-", "x" },
+  ["6"] = { "r", "w", "-" },
+  ["7"] = { "r", "w", "x" },
+}
+
+local mode_type_map = {
+  ["directory"] = "d",
+  ["link"] = "l",
+}
+
+M.mode = {
+  width = 11,
+  right_justify = true,
+  display = function(entry)
+    -- TODO: maybe just iterate over the final string?
+    local owner, group, other = string.format("%3o", entry.stat.mode):match "(.)(.)(.)$"
+    local stat = vim.tbl_flatten {
+      mode_type_map[entry.lstat.type] or "-",
+      mode_perm_map[owner],
+      mode_perm_map[group],
+      mode_perm_map[other],
+      " ",
```

Comment:
I agree with removing

```lua
entry.stat.flags ~= 0 and "@" or " "
```

since I understand from the original PR it's macOS specific and doesn't add value anyways? emacs' dired is also not showing.

Let's accordingly adjust `width = 10` and remove final `" "` completely so highlighting without the last whitespace is correct (unless there's good reason to keep it).